.github/workflows/run-tests.yml

```diff
@@ -9,57 +9,66 @@ on:
+      DATABASE_URL: "postgres://rails:password@localhost:5432/rails_test"
+      REDIS_URL: redis://localhost:6379/0
+      HOSTNAME: "http://localhost:3200"
     services:
       postgres:
         image: postgres:11-alpine
-        ports:
-          - "5432:5432"
+        ports: ["5432:5432"]
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
         env:
           POSTGRES_DB: rails_test
           POSTGRES_USER: rails
           POSTGRES_PASSWORD: password
       redis:
         image: redis
         ports: ["6379:6379"]
+        options: --entrypoint redis-server
         env:
           REDIS_HOST: redis_test
           REDIS_PORT: 6379
           REDIS_DB_ID: 0
-
-    env:
-      RAILS_ENV: test
-      DATABASE_URL: "postgres://rails:password@localhost:5432/rails_test"
-      REDIS_URL: redis://localhost:6379/0
     steps:
-      - name: Checkout code
-        uses: actions/checkout@v3
-      # Add or replace dependency steps here
-      - name: Install Ruby and gems
-        uses: ruby/setup-ruby@ee2113536afb7f793eed4ce60e8d3b26db912da4 # v1.127.0
+      - uses: actions/checkout@v3
+      - name: Install Ruby # uses .ruby-version
+        uses: ruby/setup-ruby@v1 # https://github.com/ruby/setup-ruby
+        with:
+          bundler-cache: true # runs 'bundle install' and caches installed gems automatically
+      - name: Setup Node
+        uses: actions/setup-node@v2
         with:
-          bundler-cache: true
-      # Add or replace database setup steps here
-      - name: Set up database schema
+          node-version: 16
+      - name: Install PostgreSQL
+        run: sudo apt-get -yqq install libpq-dev
+      - name: Run bundle install
+        run: |
+          gem install bundler
+          bundle install --jobs 4 --retry 3
+      - name: Setup Database
         run: bin/rails db:schema:load
-      # Add or replace test runners here
-      - name: Run tests
-        run: bin/rake
-
-  lint:
-    runs-on: ubuntu-latest
-    steps:
-      - name: Checkout code
-        uses: actions/checkout@v3
-      - name: Install Ruby and gems
-        uses: ruby/setup-ruby@ee2113536afb7f793eed4ce60e8d3b26db912da4 # v1.127.0
+      - name: Setup Google Chrome
+        run: |
+          wget https://dl.google.com/linux/direct/google-chrome-stable_current_amd64.deb
+          sudo dpkg -i google-chrome-stable_current_amd64.deb
+          sudo apt-get install -f
+      - name: setup-chromedriver
+        uses: nanasess/setup-chromedriver@v2.1.1
         with:
-          bundler-cache: true
-      # Add or replace any other lints here
-      - name: Security audit dependencies
-        run: bin/bundler-audit --update
-      - name: Security audit application code
-        run: bin/brakeman -q -w2
-      - name: Lint Ruby files
-        run: bin/rubocop --parallel
\ No newline at end of file
+          chromedriver-version: '115.0.5790.170'
```

Comment:
After much experimenting this is the only way I have found to get a specific version of chrome-driver installed in the workflow